unlock-app/src/propTypes.js

```diff
@@ -38,7 +38,7 @@ export const mechanism = PropTypes.oneOf(['0', '1', '2', undefined])
 
 export const layout = PropTypes.instanceOf(Function) //PropTypes.instanceOf(React.Component)
 
-export const locks = PropTypes.arrayOf(PropTypes.object)
```

Comment:
Actually,
Would this `export const locks = PropTypes.arrayOf(lock)` work? ( I think we define the type `lock`)

Reply:
It wouldn't right now, because it's not an array - it's an object where the properties are lock IDs. We could shift that in another PR.

Reply:
ohhh I see! no I think this is right it is better if it is an object whose keys are lock ids. sorry about that!
We can probably do something like that though:

```
PropTypes.objectOf(lock)
```